File: data/src/main/java/com/ykb/app/cryptotrader/data/dao/AuthorityDao.java
package com.ykb.app.cryptotrader.data.dao;

import com.ykb.app.cryptotrader.data.model.Authority;
import com.ykb.app.cryptotrader.data.repo.AuthorityRepo;
import org.springframework.stereotype.Component;

@Component
public final class AuthorityDao extends Dao<Authority, String> {

    public AuthorityDao(AuthorityRepo repo) {
        super(repo);
    }

}

File: library/data/data-auth/src/main/kotlin/com/ykb/app/cryptotrader/data/auth/dao/AuthorityDao.kt
package com.ykb.app.cryptotrader.data.auth.dao

import com.ykb.app.cryptotrader.data.auth.model.Authority
import com.ykb.app.cryptotrader.data.auth.repo.AuthorityRepo
import com.ykb.app.cryptotrader.data.base.Dao
import org.springframework.stereotype.Service

@Service
class AuthorityDao(repo: AuthorityRepo) : Dao<Authority, String>(repo)